From 3ab75184aa4b13f14d719994fb2d23a1b02dd918 Mon Sep 17 00:00:00 2001
From: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
Date: Fri, 21 Apr 2017 23:34:27 +0000
Subject: [PATCH 0113/1240] net: wireless: bcmdhd: make sdio_reset_comm()
 declaration weak

In case if sdio_reset_comm() is not defined in MMC driver, then
we should have a weak declartion defined in this driver to avoid
linking errors.

Change-Id: I7a5eee0148cbbd692f00a37037d3eb2acaadef9d
Signed-off-by: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
---
 drivers/net/wireless/bcmdhd/bcmsdh_sdmmc.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc.c b/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc.c
index 199522e..e1c0fb7 100644
--- a/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc.c
+++ b/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc.c
@@ -62,7 +62,10 @@ static void IRQHandler(struct sdio_func *func);
 static void IRQHandlerF2(struct sdio_func *func);
 #endif /* !defined(OOB_INTR_ONLY) */
 static int sdioh_sdmmc_get_cisaddr(sdioh_info_t *sd, uint32 regaddr);
-extern int sdio_reset_comm(struct mmc_card *card);
+int __attribute__((weak)) sdio_reset_comm(struct mmc_card *card)
+{
+	return 0;
+}
 
 #define DEFAULT_SDIO_F2_BLKSIZE		512
 #ifndef CUSTOM_SDIO_F2_BLKSIZE
-- 
2.7.4

